task/04-arrays-tasks.js

```diff
@@ -491,108 +535,130 @@ function distinct(arr) {
  *      { country: 'Belarus', city: 'Grodno' },
  *      { country: 'Belarus', city: 'Minsk' },
  *      { country: 'Poland', city: 'Lodz' }
- *     ], 
- *     item => item.country, 
+ *     ],
+ *     item => item.country,
  *     item => item.city
  *   )
- *            => 
+ *            =>
  *   Map {
  *    "Belarus" => ["Brest", "Grodno", "Minsk"],
- *    "Russia" => ["Omsk", "Samara"], 
+ *    "Russia" => ["Omsk", "Samara"],
  *    "Poland" => ["Lodz"]
  *   }
  */
 function group(array, keySelector, valueSelector) {
-   throw new Error('Not implemented');
+  let map = new Map();
+  array.map(item => {
+    if (!map.has(keySelector(item)))
+      map.set(
+        keySelector(item),
+        array
+          .filter(item2 => keySelector(item) === keySelector(item2))
+          .map(item3 => valueSelector(item3))
+      );
+  });
+  return map;
 }
 
-
 /**
  * Projects each element of the specified array to a sequence and flattens the resulting sequences into one array.
  *
  * @param {array} arr
  * @param {Function} childrenSelector, a transform function to apply to each element that returns an array of children
  * @return {array}
- * 
+ *
  * @example
  *   [[1, 2], [3, 4], [5, 6]], (x) => x     =>   [ 1, 2, 3, 4, 5, 6 ]
  *   ['one','two','three'], x=>x.split('')  =>   ['o','n','e','t','w','o','t','h','r','e','e']
  */
 function selectMany(arr, childrenSelector) {
-    throw new Error('Not implemented');
+  let result = [];
+  arr.map(item => (result = result.concat(childrenSelector(item))));
```

Comment:
JFYI: use braces in each `for`/`while`/`do ... while`/`if` operator or `callback`. You will make less mistakes.